libass/src/renderer.rs

```diff
@@ -73,16 +73,16 @@ impl<'library> Renderer<'library> {
 
     pub fn set_fonts(
         &mut self,
-        default_font: Option<&CStr>,
-        default_family: Option<&CStr>,
+        default_font: Option<&str>,
+        default_family: Option<&str>,
         default_font_provider: DefaultFontProvider,
-        fontconfig_config_path: Option<&CStr>,
+        fontconfig_config_path: Option<&str>,
         update_fontconfig_cache: bool,
     ) {
         macro_rules! unwrap_or_null {
             ($x:expr) => {
                 match $x {
-                    Some(s) => s.as_ptr(),
+                    Some(s) => CString::new(s).unwrap().as_ptr(),
```

Comment:
This is undefined behaviour, see https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html#method.as_ptr for an explanation.

You have to have every CString in a named variable for it not to be freed too early. I know it's tricky to do in this function, so I don't mind if this function has some repetitive code for it to be correct.

A way that I think should work is creating `Option<CString>` variables for each optional parameter, then using something like `var.as_deref().map(|x| x.as_ptr()).unwrap_or(ptr::null())` to get a pointer.